frameworks/MLPlan/exec.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import math
+import os
+import json
+
+from frameworks.shared.callee import call_run, result, output_subdir, utils
+
+log = logging.getLogger(__name__)
+
+def run(dataset, config):
+    log.info("\n**** ML-Plans ****\n")
+
+    is_classification = config.type == 'classification'
+    
+    # Mapping of benchmark metrics to Weka metrics
+    metrics_mapping = dict(
+        acc='ERRORRATE',
+        auc='AUC',
+        logloss='LOGLOSS',
+        f1='F1',
+        r2='R2',
+        rmse='ROOT_MEAN_SQUARED_ERROR',
+        mse='MEAN_SQUARED_ERROR',
+        rmsle='ROOT_MEAN_SQUARED_LOGARITHM_ERROR',
+        mae='MEAN_ABSOLUTE_ERROR'
+    )
+
+    metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if metric is None:
+	    raise ValueError('Performance metric {} is not supported.'.format(config.metric))
+    
+    train_file = dataset.train.path
+    test_file = dataset.test.path
+
+    training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
+    backend = config.framework_params.get('_backend', 'weka')
+
+    if backend == "weka":
+        mem_limit = str(max(config.max_mem_size_mb-1024, 2048))
+    else:
+        mem_limit = str(max(round((config.max_mem_size_mb-1024) / config.cores), 2048))
+
+    mode = backend
+    if config.type == 'regression':
+        mode += '-regression'
+
+    log.info("Running ML-Plan with backend %s in mode %s and a maximum time of %ss on %s cores with %sMB for the JVM, optimizing %s.", backend, mode, config.max_runtime_seconds, config.cores, config.max_mem_size_mb, metric)
+    log.info("Environment: %s", os.environ)
+
+    predictions_file = os.path.join(output_subdir('mlplan_out', config), 'predictions.csv')
+    statistics_file = os.path.join(output_subdir('mlplan_out', config), 'statistics.json')
+
+
+    cmd_root = "java -jar -Xmx{mem_mb}M {here}/lib/mlplan/mlplan-cli*.jar ".format(here=os.path.dirname(__file__),mem_mb=mem_limit)
+    cmd_params = dict(
+        f='"{}"'.format(train_file),
+        p='"{}"'.format(test_file),
+        t=config.max_runtime_seconds,
+        ncpus=config.cores,
+        l=metric,
+        m=mode,
+        s=config.seed,   # weka accepts only int16 as seeds
+        ooab=predictions_file,
+        os=statistics_file,
+        tmp=utils.os.TmpDir,
+        **training_params
+    )
+
+    cmd = cmd_root + ' '.join(["-{} {}".format(k, v) for k, v in cmd_params.items()])
+
+    with utils.Timer() as training:
+        utils.run_cmd(cmd, _live_output_=True)
```

Comment:
tried this locally, should work:
```suggestion
   with utils.TmpDir() as tmp_dir:
        cmd_root = "java -jar -Xmx{mem_mb}M {here}/lib/mlplan/mlplan-cli*.jar ".format(here=os.path.dirname(__file__), mem_mb=mem_limit)
        cmd_params = dict(
            f='"{}"'.format(train_file),
            p='"{}"'.format(test_file),
            t=config.max_runtime_seconds,
            ncpus=config.cores,
            l=metric,
            m=mode,
            s=config.seed,   # weka accepts only int16 as seeds
            ooab=predictions_file,
            os=statistics_file,
            tmp=tmp_dir,
            **training_params
        )

        cmd = cmd_root + ' '.join(["-{} {}".format(k, v) for k, v in cmd_params.items()])

        with utils.Timer() as training:
            utils.run_cmd(cmd, _live_output_=True)
```

Reply:
I ran the above after reinstalling ML plan, see the `-tmp` param:
```
Called ML-Plan CLI with the following params: >[-f, /Users/seb/.openml/cache/org/openml/www/datasets/61/dataset_train_0.arff, -p, /Users/seb/.openml/cache/org/openml/www/datasets/61/dataset_test_0.arff, -t, 60, -ncpus, 2, -l, LOGLOSS, -m, sklearn, -s, 312602755, -ooab, /Users/seb/repos/ml/automlbenchmark/results/mlplansklearn_test_test_local_20200807T145352/mlplan_out/iris/0/predictions.csv, -os, /Users/seb/repos/ml/automlbenchmark/results/mlplansklearn_test_test_local_20200807T145352/mlplan_out/iris/0/statistics.json, -tmp, /tmp/tmp2j03htfv]<
```
The only small annoyance is that the good old `./tmp/model_dumps` folder is still created, although remains empty.

Reply:
I reliazed the core library already provides the same feature (I reinvented the wheel, and will remove it in a next PR): https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory
So, you can use `with tempfile.TemporaryDirectory() as tmp_dir:` instead...

Reply:
Adapted the exec.py to work with the tempfile. The /tmp/model_dumps issue is also fixed now.